test/index.test.ts

```diff
@@ -650,6 +650,28 @@ describe('filter', () => {
       const actual = buildQuery({ filter });
       expect(actual).toEqual(expected);
     });
+
+    it('should handle collection operator over simple string collection with any', () => {
+      const filter = {
+        MacAddress: {
+          any: "3C:4A:92:F1:98:E2"
+        }
+      };      
+      const expected = "?$filter=MacAddress/any(macaddress: macaddress eq '3C:4A:92:F1:98:E2')";
+      const actual = buildQuery({ filter });
+      expect(actual).toEqual(expected);
+    })
+
+    it('should handle collection operator over simple string collection with all', () => {
+      const filter = {
+        MacAddress: {
+          all: "3C:4A:92:F1:98:E2"
+        }
+      };      
+      const expected = "?$filter=MacAddress/all(macaddress: macaddress ne '3C:4A:92:F1:98:E2')";
```

Comment:
Should this be `eq` instead of `ne` or am I overlooking something?

Reply:
Good question! That ```ne``` is intentional.

I could not find a good reference in the OData specification for this, because it's pretty silent when it comes to how any/all should be used against string collections, specifically.

My decision to do it this way was informed by the Azure Cognitive Search OData implementation.

They discuss the topic [here](https://docs.microsoft.com/en-us/azure/search/search-query-troubleshoot-collection-filters#rules-for-filtering-string-collections) and [here](https://docs.microsoft.com/en-us/azure/search/search-query-troubleshoot-collection-filters#common-collection-filter-errors).

![image](https://user-images.githubusercontent.com/10749074/98733674-b8875000-2355-11eb-8026-6b1cf52b1b37.png)

Hope that makes it somewhat clearer.

Reply:
Should not it be `"?$filter=not(MacAddress/all(macaddress: macaddress ne '3C:4A:92:F1:98:E2'))"` ?
if the collection is
```js
[
  {Id: 1, MacAddress: ['3C:4A:92:F1:98:E2', '00:0a:95:9d:68:16'], 
  {Id: 2, MacAddress: ['3C:4A:92:F1:98:E2'] },
  {Id: 3, MacAddress: ['00:0a:95:9d:68:16'] }
]
```
and the filter:
```js
const filter = {
    MacAddress: {
        all: "3C:4A:92:F1:98:E2"
    }
};
```

I would expect as result:
```js
[
  {Id: 2, MacAddress: ['3C:4A:92:F1:98:E2'] },
]
```
not
```js
[
  {Id: 3, MacAddress: ['00:0a:95:9d:68:16'] }
]
```

Reply:
I'm not quite following your question @jjavierdguezas.

The query you posted above returns the expected result:

![macaddr](https://user-images.githubusercontent.com/10749074/98743260-9648fe80-2364-11eb-820c-473d505b8e0d.png)

Maybe I am mistaken about your point here.

Reply:
hi @cklogs 
What I am saying is that the query should be 
`"?$filter=not(MacAddress/all(macaddress: macaddress ne '3C:4A:92:F1:98:E2'))"`
but in your tests you have
 `const expected = "?$filter=MacAddress/all(macaddress: macaddress ne '3C:4A:92:F1:98:E2')";`

(In the image you use my proposal not what you write in the tests)

do you see the difference?

Reply:
Thank you for clarifying, and I do see the difference.

My implementation was working off the assumption made by the Azure cognitive search implementation of OData. Documentation [here](https://docs.microsoft.com/en-us/azure/search/search-query-troubleshoot-collection-filters#rules-for-filtering-string-collections) and relevant line:

`The body of an any can only test for equality while the body of an all can only test for inequality.
`
To me, this means that:

1. If you make an `any` filter against a string collection, you should implicitly assume an `equality` comparison.
2. If you make an `all` filter against a string collection, you should implicitly assume a `non equality` comparison.

Perhaps my interpretation is mistaken. @jjavierdguezas What do you think?

Reply:
My interpretation of that is that you can only use those operators and you need to form the boolean expression with that restriction, but a valid (and logical) expression
to get the correct filter for `all`  using `ne` (as I see it) you need to negate the whole expression
but I could be wrong 😅

Reply:
For an user of this library it may be weird to use
```js
const filter = {
    not: {
        MacAddress: {
            all: "3C:4A:92:F1:98:E2"
        }
    }
};
```
To get as result
```js
[
  {Id: 2, MacAddress: ['3C:4A:92:F1:98:E2'] },
]
```

Reply:
I agree that may be weird. However, semantically, that expression sort of reads like:

Give me all devices for which **every single** MacAddress is **not**  '3C:4A:92:F1:98:E2'.

Wouldn't it be more natural to express this as:

Give me all devices for which **any** MacAddress is **not**  '3C:4A:92:F1:98:E2', i.e.

```
const filter = {
    not: {
        MacAddress: {
            any: "3C:4A:92:F1:98:E2"
        }
    }
};
```
or

`$filter=not(MacAddressValue/any(macaddress: macaddress eq 'C8:45:44:20:00:62'))`

Reply:
OK, I took a pencil and paper and really started thinking about this (I has been throwing here my thoughts without any formality). My wife, who was a [Logic programming](https://en.wikipedia.org/wiki/Logic_programming) teacher joined me
- **First conclusion**: what I proposed [here](https://github.com/techniq/odata-query/pull/69#discussion_r520929897) is __wrong__, negate the whole expression is not equivalent to what I want (thanks honey 😅)
- **Second conclusion**: The examples in the documentation are very simple and we don't know if we can put multiple `and` inside a `not` inside an `all`. Right know I can't imagine a real life query that we can represent with such restrictions. Here in Spain is too late in the night, so tomorrow I will keep thinking about this

I still think that the filter 
```js
const filter = {
    MacAddress: {
        all: "X"
    }
};
```
should return the items where all MacAddress are equal to `'X'`, no more no less, and the translation to OData query proposed here does not do that